keysign/send.py

```diff
@@ -92,12 +104,47 @@ def __init__(self, builder=None):
         self.notify = None
         self.internet_option = False
 
+        # Add drag and drop to the keys list widget
+        builder.connect_signals(self)
+        self.label = builder.get_object("keys_listbox")
+        self.label.drag_dest_set(Gtk.DestDefaults.ALL, [], DRAG_ACTION)
+        self.label.drag_dest_set_target_list(None)
+        self.label.drag_dest_add_text_targets()
+        self.label.drag_dest_add_uri_targets()
+
+    def on_drag_data_received(self, widget, drag_context, x, y, data, info, time):
+        if self.notify is not None:
+            log.debug("We are trying to send a key, no imports at this stage")
+            return
+
+        filename = data.get_text()
+        # If we don't have a filename it means that the user maybe dropped
+        # an attachment or an entire email.
+        if not filename:
+            filename = data.get_data().decode("utf-8")
+
+        filename = unquote(filename)
+        filename = filename[7:].strip('\r\n\x00')  # remove file://, \r\n and NULL
+        log.info("Received file: %s" % filename)
+        signatures = get_attachments(filename)
+        if not signatures:
+            with open(filename, "rb") as si:
+                signatures.append(si.read())
+
+        try:
+            for signature in signatures:
+                gpgmeh.import_signature(signature)
+            self.signature_imported()
+        except errors.GPGMEError as e:
+            log.exception(e)
```

Comment:
I *think* `log.exception` takes the Exception from the stack, so you are supposed to provide a nice string rather than the exception, e.g.  `log.exception("Could not import signatures")`.

Reply:
Right, added a comment instead of the exception itself. Btw log was smart enough to not treat "e" as a message, so the trace was not printed two times.